routing/router.go

```diff
@@ -1539,14 +1597,30 @@ func (r *ChannelRouter) SendPayment(payment *LightningPayment) ([32]byte, *route
 	// Before starting the HTLC routing attempt, we'll create a fresh
 	// payment session which will report our errors back to mission
 	// control.
-	paySession, err := r.missionControl.NewPaymentSession(
+	paySession, err := r.cfg.MissionControl.NewPaymentSession(
 		payment.RouteHints, payment.Target,
 	)
 	if err != nil {
 		return [32]byte{}, nil, err
 	}
 
-	return r.sendPayment(payment, paySession)
+	// Record this payment hash with the ControlTower, ensuring it is not
+	// already in-flight.
+	info := &channeldb.PaymentCreationInfo{
+		PaymentHash:    payment.PaymentHash,
+		Value:          payment.Amount,
+		CreationDate:   time.Now(),
+		PaymentRequest: nil,
```

Comment:
Should `PaymentRequest` be set here?

Reply:
FWIW we don't always have a payment request. But yeah in the cases that it _is_ available, it should be set here.

Reply:
It will be populated in https://github.com/lightningnetwork/lnd/pull/2717